src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -140,264 +35,133 @@ EFeatureType RandomFourierGaussPreproc::get_feature_type() {
-	return ((int32_t) dim_feature_space);
+	return m_dims;
 }
 
-void RandomFourierGaussPreproc::set_dim_feature_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_feature_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_feature_space = dim;
-
-}
-
-int32_t RandomFourierGaussPreproc::get_dim_input_space() const {
-	return ((int32_t) dim_input_space);
+void RandomFourierGaussPreproc::set_dim_feature_space(int32_t dim)
+{
+	require(m_dims > 0, "Feature space dimension must be greater than zero");
+	m_dims = dim;
 }
 
-void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ) {
-	if (kernelwidth2 <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ): kernelwidth2 <= 0 is not allowed");
-	}
-	kernelwidth=kernelwidth2;
+void RandomFourierGaussPreproc::set_kernelwidth(float64_t kernelwidth ) 
+{
+	require(m_kernelwidth > 0, "Kernelwidth must be greater than zero");
+	m_kernelwidth = kernelwidth;
 }
 
-float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
-	return (kernelwidth);
+float64_t RandomFourierGaussPreproc::get_kernelwidth() const {
+	return m_kernelwidth;
 }
 
-void RandomFourierGaussPreproc::set_dim_input_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_input_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_input_space = dim;
-
-}
-
-bool RandomFourierGaussPreproc::test_rfinited() const {
-
-	if ((dim_feature_space ==  cur_dim_feature_space)
-			&& (dim_input_space > 0) && (dim_feature_space > 0)) {
-		if ((dim_input_space == cur_dim_input_space)&&(Math::abs(kernelwidth-cur_kernelwidth)<1e-5)) {
+void RandomFourierGaussPreproc::init()
+{
+	m_kernelwidth = 1;
+	m_dims = 1000;
 
-			// already inited
-			return true;
-		} else {
-			return false;
-		}
-	}
+	m_offset = SGVector<float64_t>();
+	m_basis = SGMatrix<float64_t>();
 
-	return false;
+	SG_ADD(&m_dims, "m_dims", "Dimensionality of the output feature space.");
+	SG_ADD(&m_kernelwidth, "m_kernelwidth", "Kernel width.",
+	    ParameterProperties::HYPER);
+	SG_ADD(&m_basis, "m_basis", "Matrix of basis vectors / 1 vector per column");
+	SG_ADD(&m_offset, "m_offset", "offset vector");
 }
 
-bool RandomFourierGaussPreproc::init_randomcoefficients() {
-	if (dim_feature_space <= 0) {
-		throw ShogunException(
-				"bool CRandomFourierGaussPreproc::init_randomcoefficients(): dim_feature_space<=0 is not allowed\n");
-	}
-	if (dim_input_space <= 0) {
-		throw ShogunException(
-				"bool CRandomFourierGaussPreproc::init_randomcoefficients(): dim_input_space<=0 is not allowed\n");
-	}
-
-	if (test_rfinited()) {
-		return false;
-	}
-
-
-	io::info("initializing randomcoefficients ");
-
-	float64_t pi = 3.14159265;
+void RandomFourierGaussPreproc::generate_coeff(int32_t dim_input_space) {
```

Comment:
what about "init_random_basis" as name?

Reply:
I won't disagree but, generate_coeff sounds more intuitive to me ;)

Reply:
`init_fourier_basis` would be my favourite :)

Reply:
or `init_basis` as the base class will be called "RandomFourierFeatures" or something